Source/items.cpp

```diff
@@ -1342,19 +1343,19 @@ static bool GetItemSpace(int x, int y, int8_t inum)
 	while (rs > 0) {
 		if (itemhold[xx][yy])
 			rs--;
-		if (rs > 0) {
-			xx++;
-			if (xx == 3) {
-				xx = 0;
-				yy++;
-				if (yy == 3)
-					yy = 0;
-			}
-		}
+		if (rs <= 0)
+			continue;
+		xx++;
+		if (xx != 3)
+			continue;
+		xx = 0;
+		yy++;
+		if (yy == 3)
+			yy = 0;
 	}
 
-	xx += x - 1;
-	yy += y - 1;
+	xx += pos.x - 1;
+	yy += pos.y - 1;
 	items[inum].position = { xx, yy };

```

Comment:
Would this make sens?
```suggestion
	items[inum].position = pos + Point { xx + 1, yy - 1 };
```